src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,47 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Adds on with a by field which allows users to set a deadline.
+ *
+ * @author Yuvaraj Kumaresan
+ */
+public class Deadline extends Task {
+
+    /**
+     * Time at which the deadline is due.
+     */
+    protected LocalDateTime by;
```

Comment:
Same comment as Event variable at.
```suggestion
    protected LocalDateTime deadlineBy;
```